src/azul/__init__.py

```diff
@@ -706,9 +707,56 @@ def subscribe_to_dss(self):
     def state_machine_name(self, lambda_name):
         return config.qualified_resource_name(lambda_name)
 
+    def _concurrency(self, value: str, retry: bool) -> int:
+        """
+        >>> config._concurrency('123', False)
+        123
+        >>> config._concurrency('123', True)
+        123
+        >>> config._concurrency('123/456', False)
+        123
+        >>> config._concurrency('123/456', True)
+        456
+        >>> config._concurrency('foo', False)
+        Traceback (most recent call last):
+        ...
+        ValueError: invalid literal for int() with base 10: 'foo'
+        >>> config._concurrency('123/foo', False)
+        Traceback (most recent call last):
+        ...
+        ValueError: invalid literal for int() with base 10: 'foo'
+        >>> config._concurrency('123/', False)
+        Traceback (most recent call last):
+        ...
+        ValueError: invalid literal for int() with base 10: ''
+        >>> config._concurrency('123/456/789', False)
+        Traceback (most recent call last):
+        ...
+        ValueError: invalid literal for int() with base 10: '456/789'
+        """
+        value, sep, retry_value = value.partition('/')
+        value = int(value)
+        if sep:
+            retry_value = int(retry_value)
+        else:
+            assert not retry_value
+            retry_value = value
+        return retry_value if retry else value
+
+    def contribution_concurrency(self, *, retry: bool) -> int:
+        return self._concurrency(os.environ['AZUL_CONTRIBUTION_CONCURRENCY'], retry)
+
+    def aggregation_concurrency(self, *, retry: bool) -> int:
+        return self._concurrency(os.environ['AZUL_AGGREGATION_CONCURRENCY'], retry)
+
     @property
-    def indexer_concurrency(self):
-        return int(os.environ['AZUL_INDEXER_CONCURRENCY'])
+    def bigquery_reserved_slots(self) -> int:
+        """
+        The number of flex slots to reserve when reindexing a catalog from a
+        repository that stores its data in Google BigQuery. Slots can be
+        purchased in intervals of 100 at a rate of $0.04/slot/hour.
+        """
+        return math.ceil(self.contribution_concurrency(retry=False) / 100) * 100
```

Comment:
With this implementation, a concurrency of 199 lambdas would only get to use 100 slots. I think rounding up or down makes more sense. The price is going to change in the future. The minimum purchase is an implementation detail. I think they are phasing out the "Flex" name.

```suggestion
    def bigquery_reserved_slots(self) -> int:
        """
        The number of BigQuery slots to reserve when reindexing a catalog from a
        repository that stores its data in Google BigQuery.
        """
        # Slots must be purchased in quantities of 100
        min_slots = 100
        concurrency = self.contribution_concurrency(retry=False)
        return round(concurrency / min_slots) * min_slots
```